2023-05-08 19:41:21,854 [INFO] Node starting
2023-05-08 19:41:21,857 [INFO] Server listening on localhost:9001
2023-05-08 19:41:21,858 [INFO] Connection from ('127.0.0.1', 53242)
2023-05-08 19:41:21,859 [INFO] Connected to localhost:<socket.socket fd=768, family=2, type=1, proto=0, laddr=('127.0.0.1', 53245), raddr=('127.0.0.1', 9000)>
2023-05-08 19:41:21,859 [INFO] Connected to localhost:<socket.socket fd=792, family=2, type=1, proto=0, laddr=('127.0.0.1', 53246), raddr=('127.0.0.1', 9002)>
2023-05-08 19:41:21,859 [INFO] Connected to localhost:<socket.socket fd=828, family=2, type=1, proto=0, laddr=('127.0.0.1', 53247), raddr=('127.0.0.1', 9003)>
2023-05-08 19:41:21,860 [INFO] Connection from ('127.0.0.1', 53249)
2023-05-08 19:41:21,860 [INFO] Connection from ('127.0.0.1', 53252)
2023-05-08 19:41:23,861 [INFO] Received from ('127.0.0.1', 53242): Hello from node 0
2023-05-08 19:41:23,861 [INFO] Sending response Response: Received Hello from node 0
2023-05-08 19:41:28,862 [INFO] Received from ('127.0.0.1', 53249): Hello from node 2
2023-05-08 19:41:28,862 [INFO] Sending response Response: Received Hello from node 2
2023-05-08 19:41:38,862 [ERROR] Error handling client ('127.0.0.1', 53242): ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
2023-05-08 19:41:38,862 [ERROR] Error handling client ('127.0.0.1', 53242): ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
2023-05-08 19:41:38,862 [ERROR] Error handling client ('127.0.0.1', 53242): ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
2023-05-08 19:41:38,862 [INFO] Stopping server
2023-05-08 19:41:38,862 [ERROR] Error handling client ('127.0.0.1', 53242): ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
2023-05-08 19:41:38,864 [ERROR] Error handling client ('127.0.0.1', 53249): ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
2023-05-08 19:41:38,865 [ERROR] Connection failed first time ('127.0.0.1', 53252): RuntimeError: socket connection broken
